tracpro/contacts/fields.py

```diff
@@ -41,14 +44,75 @@ def render(self, name, value, attrs=None):
         return mark_safe(''.join(output))
 
 
+def validate_twitter_handle(handle):
+    handle_regex = r'^[a-zA-Z0-9_]{1,15}$'
+    if not re.match(handle_regex, handle):
+        raise ValidationError(
+            _("That is not a valid format Twitter handle.  A valid handle has only "
+              "letters A-Z, numbers 0-9, and underscores (_), and has up to 15 "
+              "characters."))
+
+
+def validate_numeric_twitter_id(numeric_id):
+    if not numeric_id.isdigit():  # True only if all chars are digits
+        raise ValidationError(
+            _("That is not a valid format Twitter ID.  A valid numeric ID has only "
+              "numbers 0-9."))
+
+
+def validate_twitter_id(twitter_id):
+    if '#' in twitter_id:
+        numeric_id, handle = twitter_id.split('#', 1)
+        validate_twitter_handle(handle)
+        validate_numeric_twitter_id(numeric_id)
+    else:
+        raise ValidationError(
+            _("That is not a valid format Twitter ID.  The correct format is: "
+              "<numeric id>#<Twitter handle>"))
+
+
+def validate_phone_used(number):
+    from .models import Contact
+    contacts = Contact.objects.all()
+    phone_numbers = set()
+    for contact in contacts:
+        if contact.urn.startswith('tel'):
+            phone_numbers.add(contact.urn.split(':')[1])
+
+    if number in phone_numbers:
+        raise ValidationError(
+             _("This phone number is already being used by another contact."))
+
+
+def validate_phone(number):
+    try:
+        parsed = phonenumbers.parse(number)
+    except Exception as error:
+        raise ValidationError(
+            _(error._msg))
+    else:
+        reason = phonenumbers.is_possible_number_with_reason(parsed)
+        if reason == 1:
```

Comment:
Instead of using literal numbers here, we should import names from phonenumbers and use them. That makes the code clearer and more readable, and easier to spot if we've got a mismatch between the case we're checking for and the error we're going to report.